src/theia/sfm/pose/four_point_focal_length_radial_distortion.cc

```diff
@@ -0,0 +1,632 @@
+// This file was created by Steffen Urban (urbste@googlemail.com) or company address (steffen.urban@zeiss.com)
+// December 2018
+
+
+#include <Eigen/Core>
+#include <Eigen/SVD>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+#include <Eigen/LU>
+#include <Eigen/QR>
+#include <Eigen/Cholesky>
+#include <Eigen/Eigenvalues>
+#include <random>
+
+#include "theia/util/random.h"
+#include "theia/sfm/pose/four_point_focal_length_radial_distortion.h"
+
+namespace theia
+{
+
+theia::RandomNumberGenerator rng(42);
+
+double sgn(double val) {
+  return (0.0 < val) - (val < 0.0);
+}
+
+void Solve(Eigen::Matrix<double, 64, 1> data,
```

Comment:
To help separate the pose solver logic from the groebner basis stuff, could you put the Solve method here inside a new file called p4pfr_util.h/cc and move it inside the namespace p4pfr_util